scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,227 @@
+      filenameCheck.getFailureMessage(),
+      matchedFiles,
+    )
+    return matchedFiles.toList().size != 0
+  }
+
+  /**
+   * Checks for a prohibited file content.
+   *
+   * @param repoPath the path of the repo
+   * @param searchFiles a list of all the files which needs to be checked
+   * @param fileContentCheck proto object of FileContentCheck
+   * @return file content pattern is correct or not
+   */
+  private fun checkProhibitedContent(
+    repoPath: String,
+    searchFiles: List<File>,
+    fileContentCheck: FileContentCheck,
+  ): Boolean {
+    val fileNameRegex = fileContentCheck.getFilenameRegex().toRegex()
+
+    val prohibitedContentRegex =
+      fileContentCheck.getProhibitedContentRegex().toRegex()
+
+    val matchedFiles = searchFiles.filter { file ->
+      file.name !in fileContentCheck.getExemptedFileNameList() &&
+        fileNameRegex.matches(file.name) &&
+        File(file.toString()).bufferedReader().lineSequence()
+          .foldIndexed(initial = false) { lineIndex, isFailing, lineContent ->
+            val matches = prohibitedContentRegex.matches(lineContent)
+            if (matches) {
+              logProhibitedContentFailure(
+                // Since, the line number starts from 1 and index starts
+                // from 0, therefore we have to increment index by 1, to
+                // denote the line number.
+                lineIndex + 1,
+                fileContentCheck.getFailureMessage(),
+                RepoFile.retrieveFilePath(file, repoPath)
+              )
+            }
+            isFailing || matches
+          }
+    }
+    return matchedFiles.size != 0
+  }
+
+  /**
+   * Logs the failures for filename pattern violation.
+   *
+   * @param repoPath the path of the repo to be analyzed
+   * @param errorToShow the filename error to be logged
+   * @param matchedFiles a list of all the files which had the filenaming violation
+   * @return log the filename failures
+   */
+  private fun logProhibitedFilenameFailure(
+    repoPath: String,
+    errorToShow: String,
+    matchedFiles: List<File>
+  ) {
+    if (matchedFiles.size != 0) {
+      println("Filename pattern violation: $errorToShow")
+      matchedFiles.forEach {
+        println("Prohibited file: [ROOT]/${RepoFile.retrieveFilePath(it, repoPath)}")
+      }
+      println()
+    }
+  }
+
+  /**
+   * Logs the failures for file content violation.
+   *
+   * @param lineNumberthe line number at which the failure occured
+   * @param errorToShow the failure message to be logged
+   * @param filePath the path of the file relative to the repository which failed the check
+   * @return log the prohibited content failures
+   */
+  private fun logProhibitedContentFailure(
+    lineNumber: Int,
+    errorToShow: String,
+    filePath: String
+  ) {
+    val failureMessage =
+      """
+          Prohibited content usage found on line no. $lineNumber
+          File: [ROOT]/$filePath
+          Failure message: $errorToShow
```

Comment:
Here & above: I suggest using a more compiler-esque way to reference the error by listing the file first. For example:

```
File name/path violation: <error>
- //<file_1>
- //<file_2>
...
```

And

```
//<file>:<line_number>: <error_message>
```

Reply:
Done.